src/TreeView/TreeView.tsx

```diff
@@ -36,19 +36,29 @@ export type TreeViewProps = {
 const UlBox = styled.ul<SxProp>(sx)
 
 const Root: React.FC<TreeViewProps> = ({'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledby, children}) => {
+  const rootRef = React.useRef<HTMLUListElement>(null)
   const [activeDescendant, setActiveDescendant] = React.useState('')
 
   React.useEffect(() => {
-    // Initialize the active descendant to the first item in the tree
-    if (!activeDescendant) {
-      const firstItem = document.querySelector('[role="treeitem"]')
-      if (firstItem) setActiveDescendant(firstItem.id)
+    if (rootRef.current && !activeDescendant) {
+      const currentItem = rootRef.current.querySelector('[role="treeitem"][aria-current="true"]')
+      const firstItem = rootRef.current.querySelector('[role="treeitem"]')
+
+      // If current item exists, set it as the active descendant
+      if (currentItem) {
+        setActiveDescendant(currentItem.id)
```

Comment:
Is it weird that `aria-activedescendant` updates on the parent, but `aria-current` does not move unless you manage it with state?

It seemed strange to me at first, but then I thought about it and was ok with it.

(non-blocking)

Reply:
Yeah, I think of `aria-current` more as a "default value" for `aria-activedescandant` rather than state that should be kept in sync

Reply:
I update the code comments to make that more clear 👍